tests/testthat/test-compat-dplyr.R

```diff
@@ -0,0 +1,263 @@
+# based on https://github.com/tidymodels/workflowsets/blob/main/tests/testthat/test-compat-dplyr.R
+
```

Comment:
This is one of the very _very_ few places where I moved away from completely self-contained tests in favor of skipping the entire file, since every single test here requires dplyr 1.0.0
https://github.com/tidymodels/tune/blob/b0e83a776df5779ee325938b217ce7a5192b715f/tests/testthat/test-compat-dplyr.R#L2